### ISSUE TITLE
[feature] Custom GL required class

### ISSUE THREAD

**Issue by eXponenta** [pmndrs/react-ogl#15]
Now, ogl-react resolve only prebuilded classes from
https://github.com/pmndrs/react-ogl/blob/main/src/constants.ts#L7

But for some case need has a custom class that not extends from listed, and which requires a pass gl to args.

atm this resolved as hack:

![image](https://user-images.githubusercontent.com/12572222/159897566-ef43e02d-50d9-4619-aea8-fdb0f2621a0e.png)
That  looks starange.

So, maybe we can check a static field to classes that required to construct with GL and will check it in this:
https://github.com/pmndrs/react-ogl/blob/main/src/reconciler.ts#L35

like :
```ts
if (!object && elem.requireGL || GL_ELEMENTS.some((elem) => Object.prototype.isPrototypeOf.call(elem, target) || elem === target)) {
}
```

**Comment by CodyJasonBennett** [comment]
I'm a fan of that and/or a flag for the elements themselves. TBF, GL_ELEMENTS is pretty hacky on its own to try to clean up OGL's API, but with this, we can reduce it to just updating whichever flag or prop.

**Comment by CodyJasonBennett** [comment]
What do you think of augmenting extend to handle this?

```js
extend(objects: Catalogue, requireGL: boolean)

// i.e.
extend({ CustomClass }, true)
```

**Comment by CodyJasonBennett** [comment]
Released in [`0.3.0`](https://github.com/pmndrs/react-ogl/releases/tag/v0.3.0).
